psl/schema-ast/src/ast/find_at_position/attribute.rs

```diff
@@ -1,68 +1,59 @@
+use diagnostics::Span;
+
 use crate::ast::{self};
 
-use super::{ExpressionPosition, WithSpan};
+use super::{ExpressionPosition, WithName, WithSpan};
 
 /// In an model attribute definition
 #[derive(Debug)]
 pub enum AttributePosition<'ast> {
     /// Nowhere specific inside the attribute (attribute name)
     Attribute,
     /// In an argument. (argument name)
-    Argument(&'ast str),
+    Argument(&'ast str, Span),
+    /// In the argument's name. (argument name)
+    ArgumentName(&'ast str, Span),
     /// In an argument's value. (argument name, value)
-    ArgumentValue(Option<&'ast str>, String),
+    ArgumentValue(Option<&'ast str>, String, Span),
     /// In an function argument. (function name, argument name, argument value)
-    FunctionArgument(&'ast str, &'ast str, String),
+    FunctionArgument(&'ast str, &'ast str, String, Span),
 }
 
 impl<'ast> AttributePosition<'ast> {
     pub(crate) fn new(attr: &'ast ast::Attribute, position: usize) -> Self {
-        let mut spans: Vec<(Option<&str>, ast::Span)> = attr
```

Comment:
This is no longer necessary due to the addition of a whitespace inclusive span for empty args. We can just use contains now.